src/app.service.js

```diff
@@ -335,4 +349,154 @@ export class AppService {
 
     return mainArticle.join(" ").trim().replace(/\\/g, "");
   }
+
+  gatherElementData(rootElement) {
+    const elementsData = [];
+
+    const traverse = (element, depth) => {
+      if (depth > 20 || !element) {
+        return;
+      }
+
+      const textLength = element.textContent.trim().length;
+      const allElements = element.querySelectorAll("*");
+      const linkElements = element.querySelectorAll("a");
+      const linkRatio = linkElements.length / (allElements.length + 1 || 1);
```

Comment:
```suggestion
      const linkRatio = linkElements.length / (allElements.length + 1);
```
`allElements.length + 1`이 항상 1 이상의 값을 가지는데 `|| 1`이 필요한지 모르겠습니다.
확인 부탁드려요!